pkg/utils/config.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package utils
+
+// TODO to move this to pkg
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+)
+
+var zapLoggerConfig = "zap-logger-config"
+
+// Base64ToMetricsOptions converts a json+base64 string of a
+// metrics.ExporterOptions. Returns a non-nil metrics.ExporterOptions always.
+func Base64ToMetricsOptions(base64 string) (*metrics.ExporterOptions, error) {
+	var opts metrics.ExporterOptions
+	if base64 == "" {
+		return nil, errors.New("base64 metrics string is empty")
+	}
+
+	quoted64 := strconv.Quote(string(base64))
+
+	var bytes []byte
+	// Do not care about the unmarshal error.
```

Comment:
I don't understand this comment, if the error is non-nil, we return it. So I think we care about the error?

Reply:
We are indeed caring about the error. I removed those comments.